File: snippets/typescript/snippet-09.ts
import express from 'express';
import { Glean } from '@gleanwork/api-client';

const app = express();
app.use(express.json());

const client = new Glean({
  apiToken: process.env.GLEAN_API_TOKEN!,
  serverURL: process.env.GLEAN_SERVER_URL!,
});

app.post('/api/chat', async (req, res) => {
  const { message } = req.body;
  
  const response = await client.client.chat.create({
    messages: [{ fragments: [{ text: message }] }]
  });

  const answer = response.messages
    ?.flatMap((m) => m.fragments ?? [])
    .map((f) => f.text ?? '')
    .join('') ?? '';

  res.json({ answer });
});

File: snippets/python/snippet-09.py
from fastapi import FastAPI
from glean.api_client import Glean
from pydantic import BaseModel

app = FastAPI()

class ChatRequest(BaseModel):
    message: str

@app.post("/chat")
async def chat_endpoint(request: ChatRequest):
    with Glean(
        api_token=os.getenv("GLEAN_API_TOKEN"),
        server_url=os.getenv("GLEAN_SERVER_URL"),
    ) as client:
        response = client.client.chat.create(
            messages=[{"fragments": [{"text": request.message}]}]
        )
        return {"response": str(response)}
